glue/gluedebug.h

```diff
@@ -65,41 +55,21 @@ extern int doprint_allow;
 #if STRING_IN_FLASH && !defined(USE_OPTIMIZE_PRINTF)
 #define USE_OPTIMIZE_PRINTF	// at least used in arduino/esp8266
 #endif
-// os_printf_plus() missing in osapi.h (fixed in arduino's sdk-2.1):
-//extern int os_printf_plus (const char * format, ...) __attribute__ ((format (printf, 1, 2)));
 
 #include <osapi.h> // os_printf* definitions + ICACHE_RODATA_ATTR
 
-#if UDEBUG && (UDEBUGINDEX || UDEBUGSTORE)
-// doprint() is used
-
-#undef	os_printf
-#define	os_printf(x...)		do { doprint(x); } while (0)
-
-#if STRING_IN_FLASH
-
-#define doprint(fmt, ...) \
-	do { \
-		static const char flash_str[] ICACHE_RODATA_ATTR STORE_ATTR = fmt; \
-		doprint_minus(flash_str, ##__VA_ARGS__); \
-	} while(0)
-
-#else // !STRING_IN_FLASH
-
-#define doprint(fmt, ...) doprint_minus(fmt, ##__VA_ARGS__)
-
-#endif // !STRING_IN_FLASH
-
-int doprint_minus (const char* format, ...) __attribute__ ((format (printf, 1, 2))); // format in flash
-
-#else // !( UDEBUG && (UDEBUGINDEX || UDEBUGSTORE) )
-
-#define doprint(x...)		do { os_printf(x); } while (0)
-
-#endif // !( UDEBUG && (UDEBUGINDEX || UDEBUGSTORE) )
+#if defined(ARDUINO) && !defined(LWIP14GLUE)
```

Comment:
shouldn't the `LWIP14GLUE` be applied to *everything* we build and *always* use newlib printf? i.e. 'this section only applies when we build glue, not when we include lwip headers later' (that are inconveniently chained through cc.h)
we have inconsistent `os_printf` that needs both Serial begin and debug enabled vs. `printf` that always works after Serial begin

Reply:
Done